src/globals.c

```diff
@@ -318,25 +310,48 @@ bool init_globals(void)
 		return false;
 	}
 
-/* Set configuration defaults based on options provided
-	at configure time.  */
+/* Set defaults with following order of preference:
+	1. Runtime flags
+	2. State dir configuration files
+	3. Configure time settings
+*/
+	if (!set_format_string(NULL)) {
 #ifdef FORMATID
-	set_format_string(FORMATID);
+		/* Fallback to configure time format_string if other sources fail */
+		set_format_string(FORMATID);
```

Comment:
`set_format_string()` can return false if the provided build-time value is not a positive integer. I would recommend in a future patch to cast the function call to (void) here, as it was previously, and add a configure-time check to ensure a positive integer value.